### ISSUE TITLE
Quest for glory 2 combo parser/point&click interface is broken

### ISSUE THREAD

**Issue by i30817** [adventuregamestudio/ags#1763]
**Describe the bug**
The bug was reported in scummvm: https://bugs.scummvm.org/ticket/13828

And i tested it exists in current ags master

**AGS Version**
```
***** ENGINE HAS SHUTDOWN
i3@sleipnir:/media/i3/Mordred/Games/Adventure Game Studio/Quest for Glory 2 Remake$ ags --version
Adventure Game Studio v3.6 Interpreter
Copyright (c) 1999-2011 Chris Jones and 2011-2022 others
ACI version 3.6.0.34

```

```
commit 94598f32ff20494a3239f258ac972f8108bc05bf (HEAD -> master, origin/master, origin/HEAD)
Author: Ivan Mogilko <ikm_spb@yahoo.com>
Date:   Wed Aug 31 04:53:17 2022 +0300

    Engine: improved parameter logging in SetAudioTypeVolume

```

**Game**
http://www.agdinteractive.com/games/qfg2/downloads/thegame/downloads_snd.html

**To Reproduce**
1. Start game, create a new character, and chose the combined parser/p&c interface
2. click the lips icon on the main character
3. see that the last option, which is supposed to show a text field to insert the text doesn't.

**Expected behavior**
The image on the scummvm bug report shows it.

**Comment by i30817** [comment]
I'm doing a bisect, i'll have a commit for you 'soon'.

**Comment by i30817** [comment]
```
9e9c8ffafe465b907ed0d22bcb1ba4df06803b8b is the first bad commit
commit 9e9c8ffafe465b907ed0d22bcb1ba4df06803b8b
Author: Ivan Mogilko <ikm_spb@yahoo.com>
Date:   Tue Aug 10 23:25:01 2021 +0300

    Common: gui controls draw themselves at requested offset

 Common/gui/guibutton.cpp  | 43 ++++++++++++++++++++++---------------------
 Common/gui/guibutton.h    |  8 ++++----
 Common/gui/guiinv.h       |  2 +-
 Common/gui/guilabel.cpp   |  8 ++++----
 Common/gui/guilabel.h     |  2 +-
 Common/gui/guilistbox.cpp | 28 ++++++++++++++--------------
 Common/gui/guilistbox.h   |  2 +-
 Common/gui/guimain.cpp    |  2 +-
 Common/gui/guiobject.h    |  2 +-
 Common/gui/guislider.cpp  | 10 +++++-----
 Common/gui/guislider.h    |  2 +-
 Common/gui/guitextbox.cpp |  8 ++++----
 Common/gui/guitextbox.h   |  4 ++--
 Engine/ac/guiinv.cpp      | 10 +++++-----
 Engine/gui/gui_engine.cpp |  8 ++++----
 15 files changed, 70 insertions(+), 69 deletions(-)
```


I had to skip a lot of steps on the bisect, with git bisect skip because when i tried to trigger the dialog menu, it froze the gui (still could move the cursor but it changed to default and all animations stopped).

I hope i did the bisect right.


Another bug i noticed in this game in main is if you start the game and try to load a game (without any save) it's show as the first entry 'invalid slot 0'. Not sure it's related.

**Comment by i30817** [comment]
That other bug:
```
17ea3efe539fa9694e91e0c52b4be4b2c1ab8134 is the first bad commit
commit 17ea3efe539fa9694e91e0c52b4be4b2c1ab8134
Author: Ivan Mogilko <ikm_spb@yahoo.com>
Date:   Fri Apr 15 02:59:58 2022 +0300

    Linux: fixed FindFile works only in cwd

 Common/util/directory.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

```

**Comment by ivan-mogilko** [comment]
Fixed the primary bug: 394866b2928433475fcfd0a3ab298e2b70570cbc

I made a wrong program design choice with these old commits (the main reason for that change was 5cbe36626842ddd900cbda8fd4b6ddfc3804456f), should have used some semi-automatic approach, where drawing surface has a "draw offset" saved and aligns all operations with it, instead of telling each gui control where to draw.

**Comment by i30817** [comment]
Both fixed.
